### PR TITLE
Logging, style, docstrings, code review!

### PR DESCRIPTION
**Pull request by erinspace**

### CODE REVIEW

**Comment by fabianvf** [comment]
Looks good to me. @chrisseto second pass?

File: scrapi/base/__init__.py [hunk 1]
```diff
@@ -110,26 +136,29 @@ def get_contributors(self, result):
         or creators in the OAI namespaces """
 
         contributors = result.xpath(
-            '//dc:contributor/node()', namespaces=self.NAMESPACES) or ['']
```
Comment (chrisseto) [inline scrapi/base/__init__.py]: `['']` is a list with a singular empty string in it where `[]` is just an empty list.
Was this change intended?

File: scrapi/base/__init__.py [hunk 2]
```diff
@@ -110,26 +136,29 @@ def get_contributors(self, result):
...
+            '//dc:contributor/node()',
+            namespaces=self.NAMESPACES
+        ) or []
         creators = result.xpath(
-            '//dc:creator/node()', namespaces=self.NAMESPACES) or ['']
+            '//dc:creator/node()',
+            namespaces=self.NAMESPACES
+        ) or []
```
Comment (chrisseto) [inline scrapi/base/__init__.py]: See above comment.

File: scrapi/base/__init__.py [hunk 3]
```diff
@@ -110,26 +136,29 @@ def get_contributors(self, result):
...
 
         all_contributors = contributors + creators
 
         contributor_list = []
         for person in all_contributors:
-            if person:
```
Comment (chrisseto) [inline scrapi/base/__init__.py]: Why was this null check removed?

File: scrapi/base/__init__.py [hunk 4]
```diff
@@ -171,36 +204,61 @@ def get_properties(self, result, property_list):
 
         properties = {}
         for item in property_list:
-            prop = (result.xpath('//dc:{}/node()'.format(item), namespaces=self.NAMESPACES) or [''])
-            prop += (result.xpath('//ns0:{}/node()'.format(item), namespaces=self.NAMESPACES) or [''])
-
-            properties[item] = prop if len(prop) > 1 else prop[0]
+            prop = (
```
Comment (chrisseto) [inline scrapi/base/__init__.py]: Can you remove the surrounding `()` it makes the value look like it should be a tuple when its not.

File: scrapi/base/__init__.py [hunk 5]
```diff
@@ -171,36 +204,61 @@ def get_properties(self, result, property_list):
...
+                result.xpath(
+                    '//dc:{}/node()'.format(item),
+                    namespaces=self.NAMESPACES
+                ) or ['']
+            )
+            prop += (
```
Comment (chrisseto) [inline scrapi/base/__init__.py]: See above comment.

File: scrapi/base/__init__.py [hunk 6]
```diff
@@ -171,36 +204,61 @@ def get_properties(self, result, property_list):
...
+                result.xpath(
+                    '//ns0:{}/node()'.format(item),
+                    namespaces=self.NAMESPACES
+                ) or ['']
+            )
+
+            if len(prop) > 1:
+                properties[item] = [self.copy_to_unicode(item) for item in prop]
+            else:
+                properties[item] = self.copy_to_unicode(prop[0])
```
Comment (chrisseto) [inline scrapi/base/__init__.py]: This will make `properties[item]` a string rather than a list, is this intended?

File: scrapi/base/__init__.py [hunk 7]
```diff
@@ -171,36 +204,61 @@ def get_properties(self, result, property_list):
...
-        return self.copy_to_unicode(title)
+        title = result.xpath(
+            '//dc:title/node()',
+            namespaces=self.NAMESPACES)
+        return self.copy_to_unicode(title[0])
 
     def get_description(self, result):
-        description = (result.xpath('//dc:description/node()', namespaces=self.NAMESPACES) or [''])[0]
-        return self.copy_to_unicode(description)
+        description = (
```
Comment (chrisseto) [inline scrapi/base/__init__.py]: Same comment about tuples here

File: scrapi/base/__init__.py (hunk 1 above)
Comment (erinspace) [inline scrapi/base/__init__.py]: Right! Good point, removed, because xpath returns just an empty list with no elements found anyhow. Changed to simplify for loop below.

File: scrapi/base/__init__.py (hunk 3 above)
Comment (erinspace) [inline scrapi/base/__init__.py]: Since the list will just be empty if there are no contributors, no more need for the null check.

File: scrapi/base/__init__.py (hunk 4 above)
Comment (erinspace) [inline scrapi/base/__init__.py]: Right! Updated other things but forgot to remove the (), thank you.

File: scrapi/base/__init__.py (hunk 6 above)
Comment (erinspace) [inline scrapi/base/__init__.py]: Yep! Wanted to have singular items show up as just one instead of one item inside a list - this is done to make things as generic as possible, and because items that are singular - publisher, type, etc, will always have just one element, and it'd be strange for them to show up as just one element in a list. 

We can discuss this though!

**Comment by fabianvf** [comment]
@chrisseto, I'm going to merge, let me know if I shouldn't
